### PR TITLE
[Evaluation] [Refac] Made 'CkValue' store actual constants in 'VCon'

### PR DESCRIPTION
**Pull request by effectfully**
This does for the CK machine what was done for the CEK machine long before.

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `$(nix-build default.nix -A pkgsLocal.updateMaterialized)` to update the materialized Nix files
   - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
   - [ ] `$(nix-shell shell.nix --run fix-stylish-haskell)` to fix any formatting issues
- If you changed any Purescript files:
   - [ ] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs [hunk 1]
```diff
@@ -143,19 +143,19 @@ failure into a 'Term', apart from the straightforward generalization of 'CekM'.
 -}
 
 -- | The CEK machine-specific 'EvaluationException', parameterized over @term@.
-type CekEvaluationExceptionCarrying fun term =
+type CekEvaluationExceptionCarrying term fun =
```
Comment (effectfully) [inline plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs:146]: Changed that for consistency (in `CekCarryingM` `term` comes before `fun`).

File: plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs [hunk 2]
```diff
@@ -65,8 +66,8 @@ data CkValue uni fun =
    case anyway (see Note [Ignoring context in UserEvaluationError] in
    Exception.hs)
 -}
-mkBuiltinApplication :: () -> fun -> Arity -> [Type TyName uni ()] -> [Term TyName Name uni fun ()] -> Term TyName Name uni fun ()
-mkBuiltinApplication () bn arity0 tys0 args0 =
+mkBuiltinApplication :: fun -> Arity -> [Type TyName uni ()] -> [Term TyName Name uni fun ()] -> Term TyName Name uni fun ()
+mkBuiltinApplication bn arity0 tys0 args0 =
```
Comment (effectfully) [inline plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs:70]: Removed all `()` arguments from functions. I can bring them back if anybody cares. I don't care either way, so went for "less code to think about".

File: plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs [hunk 3]
```diff
@@ -77,38 +78,45 @@ mkBuiltinApplication () bn arity0 tys0 args0 =
...
 
 data CkUserError =
     CkEvaluationFailure -- Error has been called or a builtin application has failed
     deriving (Show, Eq)
 
+-- | The CK machine-specific 'EvaluationException' parameterized over @term@.
+type CkEvaluationExceptionCarrying term fun =
+    EvaluationException CkUserError fun term
+
+-- See Note [Being generic over @term@ in 'CekM']
```
Comment (effectfully) [inline plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs:107]: The note is kinda big, so I don't want to replicate it in this file, given that it applies to the CK machine the same way it applies to the CEK one.

File: plutus-core/test/Evaluation/Golden/polyErrorInst.plc.golden [hunk 4]
```diff
@@ -1,2 +1,3 @@
 (Left An error has occurred:  User error:
-The provided Plutus code called 'error'.)
\ No newline at end of file
+The provided Plutus code called 'error'.
+Caused by: (error a_0))
```
Comment (effectfully) [inline plutus-core/test/Evaluation/Golden/polyErrorInst.plc.golden:3]: I admit the error is rather useless. Although if you have multiple calls to `error` in a program, having an uninstatiated type information is better than nothing.

@michaelpj @kwxm are we going to drop the typed CEK machine some day? I really dislike those uninstated types (and updating two machines all the time). In my view the typed CEK machine is completely redundant and can be dropped right away.

Comment (michaelpj) [reply plutus-core/test/Evaluation/Golden/polyErrorInst.plc.golden:3]: I think we could probably drop the typed CK machine. We have a *lot* of machines at the moment, especially if we're duplicating them for TPLC and UPLC.

**Review by michaelpj — APPROVED** [review]

**Comment by michaelpj** [comment]
One test needs updating.

**Comment by effectfully** [comment]
> One test needs updating.

It's a bit more than that. The CK machine handles types properly and the CEK machine doesn't, so we get two different results, which is a problem when you have a single golden file. I removed printing the cause of `Error`, now we get the same result for both the machines again.

**Pull request merged**